### PR TITLE
simplify loops in callfunc()

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/e2ir.d [hunk 1]
```diff
@@ -188,48 +187,55 @@ private elem *callfunc(const ref Loc loc,
...
 
-            size_t nparams = tf.parameterList.length;
-            if (i - j < nparams && i >= j)
+            if (i - j < tf.parameterList.length &&
+                i >= j &&
+                tf.parameterList[i - j].storageClass & (STC.out_ | STC.ref_))
             {
-                Parameter p = tf.parameterList[i - j];
-
-                if (p.storageClass & (STC.out_ | STC.ref_))
```
Comment (thewilsonator) [inline src/dmd/e2ir.d]: What happened to this?

Comment (WalterBright) [reply src/dmd/e2ir.d]: It's now line 215 on the green side.

**Review by thewilsonator — APPROVED** [review]

**Comment by WalterBright** [comment]
The buildkite/dmd failure looks irrelevant.

**Comment by thewilsonator** [comment]
Yep, it is.

**Pull request merged**